demisto_sdk/commands/common/hook_validations/incident_type.py

```diff
@@ -0,0 +1,127 @@
+from distutils.version import LooseVersion
+
+from demisto_sdk.commands.common.constants import Errors
+from demisto_sdk.commands.common.hook_validations.base_validator import BaseValidator
+from demisto_sdk.commands.common.tools import print_error
+
+
+class IncidentTypeValidator(BaseValidator):
+    """IncidentTypeValidator is designed to validate the correctness of the file structure we enter to content repo.
+    And also try to catch possible Backward compatibility breaks due to the performed changes.
+    """
+
+    def is_backward_compatible(self):
+        """Check whether the Incident Type is backward compatible or not
+        """
+        if not self.old_file:
+            return True
+
+        is_bc_broke = any(
+            [
+                self.is_changed_from_version()
+            ]
+        )
+
+        return not is_bc_broke
+
+    def is_valid_incident_type(self):
+        """Check whether the Incident Type is valid or not
+        """
+        is_incident_type__valid = all([
+            self.is_valid_version()
+        ])
+
+        # check only on added files
+        if not self.old_file:
+            is_incident_type__valid = all([
+                is_incident_type__valid,
+                self.is_id_equals_name(),
+                self.is_including_fields()
+            ])
+
+        return is_incident_type__valid
+
+    def is_valid_version(self):
+        # type: () -> bool
+        """Check if a valid version.
+        Returns:
+            bool. Whether the version is valid or not.
+        """
+        return super(IncidentTypeValidator, self)._is_valid_version()
+
+    def is_current_valid_from_version(self):
+        # type: () -> bool
+        """Check if the current file fromversion is valid.
+        Returns:
+            bool. Whether the current fromversion is valid or not.
+        """
+        error_msg = None
```

Comment:
why do you need this var? isn't it better to just print any error that you find?